DNN Platform/Library/Services/Scheduling/Scheduler.cs

```diff
@@ -128,6 +128,12 @@ public static bool ValidateServersAreActiveForScheduledItem(ScheduleItem schedul
                 // Get the enabled servers with recent activity.
                 var enabledServers = ServerController.GetEnabledServersWithActivity();
 
+                if (!enabledServers.Any())

```

Comment:
This change breaks the desired behavior here to move the job to another server that has activity

Reply:
Hmm, the intention is to fix the exception that occurs on `enabledServers.First()` when there are no enabledServers:


```
                // Get the enabled servers with recent activity.
                var enabledServers = ServerController.GetEnabledServersWithActivity();
```
if enabledServers is an empty `List<ServerInfo>` here.
```
                // Validate that any server in the scheduled item is in the enabled server list with activity in the last 10 minutes.
                bool serverHasRecentActivity = enabledServers.Any(i => servers.Contains(i.ServerName.ToLowerInvariant()));
```
`serverHasRecentActivity` will be `false` here.
```
                // If no servers in the scheduled item had recent activity, assign this scheduled item to the server with the most recent activity
                if (!serverHasRecentActivity)
                {
```
then this line will throw a `InvalidOperationException: Sequence contains no elements` because there is no `.First()`.
```
                    // Assign a new server to this task then run it
                    scheduleItem.Servers = $",{enabledServers.First().ServerName.ToLowerInvariant()},";

                    // Save the scheduled item back to the database.
                    SchedulingController.UpdateSchedule(scheduleItem);
                    return false;
                }

```

That's what we saw in the logs and what I tried to fix here.

Reply:
Ah this makes sense but we don't want to just run everywhere

Reply:
The way I see it, I'm not changing behavior in any way. I'm just preventing an exception by saying: if there are no available servers, then do nothing. 

That said, I'm happy to improve something if I know what to improve...

Reply:
@mtrutledge Do you have any guidance on this one

Reply:
There should always be at least 1 enabled server. In what conditions are we seeing the WebServers table and the enabledServers count is 0?

Reply:
Sorry for the delay,I had to dive into that. 

We experienced that after changing the frequency of the Montior Web Servers task. It looks like if that task isn't running (or not frequently enough?) it might occur that there are no enabled servers and the exception that I'm catching here, is thrown.